subprojects/jacoco/src/main/java/org/gradle/testing/jacoco/plugins/JacocoReportAggregationPlugin.java

```diff
@@ -91,7 +91,6 @@ public void apply(Project project) {
         });
 
         ArtifactView classDirectories = codeCoverageResultsConf.getIncoming().artifactView(view -> {
-            view.withVariantReselection();
```

Comment:
Is this an indication that we have an issue with variant re-selection and capabilities?

Reply:
There is an issue here. Reselection will reselect "out of" your capability. Reselection should be restricted to only variants with the same capabilities as the original.

Reply:
Reselection ignores the attributes set on the `Configuration` that created the `ArtifactView`.  We need to maintain those attributes here to narrow selection to the single proper variant, otherwise there are multiple potential matches with `classes` or compatible with `classes` and selection fails.

Reply:
The capabilites issue exists. But it may not be related to this issue

Reply:
The multiple potential matches being from the production compiled code and the test fixtures code? I think capabilities may be related since both variants should have identical attributes but just different capabilities

Reply:
Yeah, that's what I think is going on too.  We're re-selecting from the full set of variants instead of the ones for the requested capability. 

This is good enough for 8.2.1, but there's something deeper here that's missing.